code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -276,10 +343,76 @@
 	max_energy = 100
 	desc = "A technological marvel, supposedly able to mix just the mixture you'd like to drink the moment you ask for one."
 	dispensable_reagents = list("ice", "cream", "cider", "beer", "kahlua", "whiskey", "wine", "vodka", "gin", "rum", "tequila", "vermouth", "cognac", "ale", "mead", "synthanol")
+	/var/list/special_reagents3 = list(list("iced_beer"),
+								list("irishcream", "manhattan",),
+								list("antihol", "synthignon", "bravebull"))
 	hack_message = "You disable the 'nanotrasen-are-cheap-bastards' lock, enabling hidden and very expensive boozes."
 	unhack_message = "You re-enable the 'nanotrasen-are-cheap-bastards' lock, disabling hidden and very expensive boozes."
 	hacked_reagents = list("goldschlager", "patron", "absinthe", "ethanol", "nothing", "sake")
 
+/obj/machinery/chem_dispenser/beer/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/beer(null)
+	component_parts += new /obj/item/stock_parts/matter_bin(null)
+	component_parts += new /obj/item/stock_parts/matter_bin(null)
+	component_parts += new /obj/item/stock_parts/manipulator(null)
+	component_parts += new /obj/item/stock_parts/capacitor(null)
+	component_parts += new /obj/item/stock_parts/console_screen(null)
+	component_parts += new /obj/item/stock_parts/cell/super(null)
+	RefreshParts()
+
+/obj/machinery/chem_dispenser/beer/upgraded/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/beer(null)
+	component_parts += new /obj/item/stock_parts/matter_bin/super(null)
+	component_parts += new /obj/item/stock_parts/matter_bin/super(null)
+	component_parts += new /obj/item/stock_parts/manipulator/pico(null)
+	component_parts += new /obj/item/stock_parts/capacitor/super(null)
+	component_parts += new /obj/item/stock_parts/console_screen(null)
+	component_parts += new /obj/item/stock_parts/cell/hyper(null)
+	RefreshParts()
+
+/obj/machinery/chem_dispenser/beer/RefreshParts()
+	var/i
+	for(var/obj/item/stock_parts/manipulator/M in component_parts)
+		for(i=1, i<=M.rating, i++)
+			dispensable_reagents = sortList(dispensable_reagents | special_reagents3[i])
+
+/obj/machinery/chem_dispenser/beer/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/reagent_containers/glass))
+		if(panel_open)
+			to_chat(user, "<span class='notice'>Close the maintenance panel first.</span>")
+			return
+		..()
```

Comment:
need to return the parent values in attackby, not just call parent.

Reply:
This is exactly how the chem dispenser did it, not sure what you want me to do here/why it's a bad thing?

Reply:
Just because one section of the code has it one way doesn't mean that the proper way to go about it.

Everything needs to actively tie into the object damage system one day, and therefore, everything in attackby needs to return the value of the parent proc. You need do do this for anything that you add---and if you add anything that builds on something else that doens't do it ,then you're going to have to make *it* return the proper value and test, in game, if it actually works or not.

Reply:
Okay, so how do I go about that? This is the first time I've worked with doing something like this in code. Not sure what else uses the same system or I'd look for it myself.